www/webfilter/src/opnsense/mvc/app/views/OPNsense/WebFilter/index.volt

```diff
@@ -0,0 +1,268 @@
+        if (data['status'] != 'done') {
+            if (data['status'] == 'running') {
+                $("#dl_categ_progress").addClass("fa fa-spinner fa-pulse");
+                $("#dl_categ").prop("disabled", true);
+                setTimeout(timeoutCheck, 2500);
+            }
+            $('#dl_categ_alert').show();
+        }
+    });
+
+        // Reconfigure webfilter - activate changes
+        $('[id*="reconfigureAct"]').each(function(){
+            $(this).click(function(){
+                // set progress animation
+                $('[id*="reconfigureAct_progress"]').each(function(){
+                    $(this).addClass("fa fa-spinner fa-pulse");
+                });
+
+            });
+        });
+        
+        // form save event handlers for all defined forms
+        $('[id*="save_"]').each(function(){
+            $(this).click(function(){
+                var frm_id = $(this).closest("form").attr("id");
+                var frm_title = $(this).closest("form").attr("data-title");
+
+                // set progress animation
+                $("#"+frm_id+"_progress").addClass("fa fa-spinner fa-pulse");
+
+                // save data for tab
+                saveFormToEndpoint(url="/api/webfilter/settings/set",formid=frm_id,callback_ok=function(){
+
+                    // on correct save, perform reconfigure
+                    ajaxCall(url="/api/webfilter/service/reconfigure", sendData={}, callback=function(data,status) {
+                        if (status != "success" || data['status'] != 'ok') {
+                            BootstrapDialog.show({
+                                type: BootstrapDialog.TYPE_WARNING,
+                                title: "{{ lang._('Error reconfiguring WebFilter') }}",
+                                message: data['status'],
+                                draggable: true
+                            });
+                        } else {
+                            ajaxCall(url="/api/webfilter/service/status", sendData={}, callback=function(data,status) {
+                                updateServiceStatusUI(data['status']);
+                            });
+                        }
+                        // when done, disable progress animation.
+                        $("#"+frm_id+"_progress").removeClass("fa fa-spinner fa-pulse");
+                    });
+
+                });
+            });
+        });
+
+    $("#dl_categ").click(function(){
+        $("#dl_categ_progress").addClass("fa fa-spinner fa-pulse");
+        $("#dl_categ").prop("disabled", true);
+        ajaxCall(url="/api/webfilter/service/download", sendData={action:1}, callback_ok=function(){
+            setTimeout(timeoutCheck, 2500);
+        });
+    });
+
+        // update history on tab state and implement navigation
+        if(window.location.hash != "") {
+            $('a[href="' + window.location.hash + '"]').click()
```

Comment:
this may allow XSS - needs to be tested with a hash like `#abc"],custom_selector,[href="abc`. Not sure if the browser will accept that.

Reply:
I need a professional help here! ;-)

Reply:
https://api.jquery.com/jQuery.escapeSelector/

Reply:
Same here... I used this code from another plugins. I'm not a java guy... help from someone with a better javascript knowledge will be appreciated!